src/msgs/data_requests/execute/reveal_result.rs

```diff
@@ -12,6 +12,71 @@ pub struct Execute {
     pub stdout:      Vec<String>,
 }
 
+impl SignSelf for Execute {
+    type Extra = (u64, Hash);
+
+    fn proof(&self) -> Result<Vec<u8>> {
+        Ok(hex::decode(&self.proof)?)
+    }
+
+    fn msg_hash(
+        &self,
+        chain_id: &str,
+        contract_addr: &str,
+        (dr_height, reveal_body_hash): Self::Extra,
+    ) -> Result<Hash> {
+        Ok(hash([
+            "reveal_data_result".as_bytes(),
+            self.dr_id.as_bytes(),
+            &dr_height.to_be_bytes(),
+            &reveal_body_hash,
+            chain_id.as_bytes(),
+            contract_addr.as_bytes(),
+        ]))
+    }
+}
+
+impl Execute {
+    #[allow(clippy::too_many_arguments)]
+    pub fn new(
+        dr_id: String,
+        reveal_body: RevealBody,
+        public_key: String,
+        stderr: Vec<String>,
+        stdout: Vec<String>,
```

Comment:
I was wondering if there should be any consideration not including these 2 fields in the signature. Any thoughts?

Reply:
The concern is about replaying the reveal transaction... but let's think about this.

Maybe they could be included as a sort of hash but inside of the reveal body.

Reply:
If you want to merge the PR without this, we should create an issue. :)